ethcore/src/block.rs

```diff
@@ -498,7 +498,7 @@ pub fn enact(
 	tracing: bool,
 	db: Box<JournalDB>,
 	parent: &Header,
-	last_hashes: LastHashes,
+	last_hashes: Arc<LastHashes>,
```

Comment:
Changing `EnvInfo` to hold a reference is much more involved. I'll leave it to future PR.